src/sdk/index.js

```diff
@@ -4,23 +4,42 @@ import {merge} from 'lodash'
 import api from './api'
 import {DEFAULT_HEADERS} from './headers'
 
+function configGlobalSpec (globalSpec) {
+
+  let headers
+  if (typeof globalSpec.headers === 'function') {
+    headers = {...globalSpec.headers()}
```

Comment:
The issue is that the way merge is working, there are some default header fields in `defaultSpec` that are lost when `globalSpec.headers` is a function. The spreads in the `configGlobalSpec` seem to be redundant actually and could be removed, but an object must always be returned.